### ISSUE TITLE
Dynamic ads

### ISSUE THREAD

**Issue by jpgill86** [critrolesync/critrolesync.github.io#6]
Beginning in early August 2020 (08c06a6), it appears that the Critical Role podcast feed (Campaign 2, C2E20 and later) has switched hosting and distribution from Podbean to Anchor Podcasts. Anchor allows for [dynamic advertisement insertion](https://help.anchor.fm/hc/en-us/articles/360020241332-Step-by-Step-guide-to-setting-up-Sponsorships), where the ad that plays depends on variables such as which advertiser is willing to pay the most at any given time.

For the purposes of this project, this is bad news, since this means that the duration of ad segments could change frequently. This may make it impossible to provide precise timestamp mapping using our current techniques.

One sophisticated solution to this problem would be to use an automated tool (#7) periodically, say once per month, to re-catalog all episodes. Alternatively, if the ad campaigns change uniformly across episodes and only rarely, it may be possible to make corrections manually. However, an automated approach sounds much less labor intensive to me!

Edit: Beginning with C2E105, midroll ads have started appearing in the podcasts at the break. [This first one](https://podcasts.google.com/?feed=aHR0cHM6Ly9jcml0cm9sZS5wb2RiZWFuLmNvbS9mZWVkLnhtbA&episode=YjJmODU0ODgtYjBiZC00OWY3LWIxNWUtYzJlMWU5MTM2YzY3&pe=1&pep=9510000) is for Anchor itself! As of 2020-08-13, I am not aware of these being inserted retroactively into old episodes.

**Comment by jpgill86** [comment]
As I anticipated, this week all previously released episodes from this feed (C2E20 and later) were changed retroactively by adding new ads ([see feed diff](https://www.diffchecker.com/GuhB3AA3)).

The good news is that it appears (at first glance) that the same 60-second midroll Anchor ad was added to all of them. This means that timestamps can be corrected by adding a constant.

The re-release of these podcast episodes also included a file type conversion (from MP3 to M4A), which appears (again, at first glance) to have eliminated the playback rate discrepancies described in #5 between Chrome and Firefox (and hopefully all podcast players). This also breaks all timestamps for Chrome/Android, but again this can be corrected by multiplying by a constant.
